### ISSUE TITLE
Bug: [no-unnecessary-condition] not work when use object field

### ISSUE THREAD

**Issue by njzydark** [typescript-eslint/typescript-eslint#6614]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=4.9.5&jsx=true&sourceType=module&code=MYewdgzgLgBCBGArGBeGBvAUDGBDAXDAIyYC+mmAlgGYwAUCiAdLqimoywJQbYyiQQAGwCmTISADmdAOQATajK5kgA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK+SpPRRE0aB2iRwYAL4gtQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3QAacDToMmeCmFHopAXxAqgA

### Repro Code

```typescript
const obj = {
  a: 1
}

if (obj.a === obj.a) {
  console.log('df')
}
```

### ESLint Config

```javascript
module.exports = {
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": "error"
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true,
    "strict": true
  }
}
```

### Expected Result

`obj.a === obj.a` will throw a eslint error

### Actual Result

`obj.a === obj.a` no eslint error

**Comment by bradzacher** [comment]
This is working as expected. The rule looks at your types to determine if the check is unnecessary.
As far as your types are concerned - you're comparing a `number` and a `number` - which obviously is not unnecessary.

We also cannot report on this case because we don't know if you've defined a `get` for that property or not. Eg your code might instead be this, which is not an unnecessary condition!

```ts
let i = 0;
const x = {
  get prop() {
    if (random() > 0.5) { i++ }
    return i; 
  },
};

if (x.prop === x.prop) {} // not unnecessary!!!
```

**Comment by njzydark** [comment]
> This is working as expected. The rule looks at your types to determine if the check is unnecessary. As far as your types are concerned - you're comparing a `number` and a `number` - which obviously is not unnecessary.

Can't we additionally check the variable names for consistency before making a final decision on whether they are non-essential in the case of consistent base types?
